scripts/prepare-crds

```diff
@@ -65,18 +65,13 @@ name: ${name}-crd
 type: application
 EOF
 
-# Add providesGVR annotation
-providesGVR="$(cat ${f}/package.yaml | yq r - generateCRDChart.providesGVR)"
-if ! [[ -z ${providesGVR} ]]; then
-	if [[ -z "$(yq r ${f}/charts/Chart.yaml 'annotations[catalog.cattle.io/auto-install-gvr]')" ]]; then
-		yq w -i ${f}/charts/Chart.yaml "annotations[catalog.cattle.io/auto-install-gvr]" "${providesGVR}"
-	fi
-	if [[ -z "$(yq r ${f}/charts/Chart.yaml 'annotations[catalog.cattle.io/provides-gvr]')" ]]; then
-		yq w -i ${f}/charts-crd/Chart.yaml "annotations[catalog.cattle.io/provides-gvr]" "${providesGVR}"
-	fi
-	if [[ -z "$(yq r ${f}/charts/Chart.yaml 'annotations[catalog.cattle.io/hidden]')" ]]; then
-		yq w -i ${f}/charts-crd/Chart.yaml "annotations[catalog.cattle.io/hidden]" "true"
-	fi
+# Make the primary chart auto-install the CRD chart
+if [[ -z "$(yq r ${f}/charts/Chart.yaml 'annotations[catalog.cattle.io/auto-install]')" ]]; then
+	yq w -i ${f}/charts/Chart.yaml "annotations[catalog.cattle.io/auto-install]" "${name}-crd=match"
+fi
+# Hide the CRD chart
+if [[ -z "$(yq r ${f}/charts-crd/Chart.yaml 'annotations[catalog.cattle.io/hidden]')" ]]; then
+	yq w -i ${f}/charts-crd/Chart.yaml "annotations[catalog.cattle.io/hidden]" "true"
```

Comment:
If we can merge this PR with https://github.com/rancher/charts/pull/595/commits/80d71abe311594a5f235f2987a032528188690a6, this can just be directly added into the `chart-template/chart-crd-base/Chart.yaml` instead of introducing a new if condition.

Reply:
We can handle that later / separately. This is a blocker for UI so need to get it moving, if we handle in both places thats ok.